src/Agent.Listener/Configuration/FeatureFlagProvider.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Runtime.Serialization;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using YamlDotNet.Core.Tokens;
+
+namespace Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(FeatureFlagProvider))]
+    public interface IFeatureFlagProvider : IAgentService
+    {
+        /// <summary>
+        /// Gets the status of a feature flag from the specified service endpoint.
+        /// If request fails, the feature flag is assumed to be off.
+        /// </summary>
+        /// <param name="serviceEndpoint">The service endpoint to get the feature flag status from.</param>
+        /// <param name="featureFlagName">The name of the feature flag to get the status of.</param>
+        /// <returns>The status of the feature flag.</returns>
+        public Task<FeatureFlagStatus> GetFeatureFlagAsync(ServiceEndpoint serviceEndpoint, string FeatureFlagName);
+    }
+    /// <summary>
+    /// 
+    /// </summary>
+    public class FeatureFlagProvider : AgentService, IFeatureFlagProvider
+    {
+        private HttpClient httpClient { get; set; }
+
+        public FeatureFlagProvider()
+        {
+            httpClient = new HttpClient();
+            httpClient.Timeout = TimeSpan.FromSeconds(10);
+        }
+        public FeatureFlagProvider(HttpClient client) : this()
+        {
+            httpClient = client;
+        }
+
+        public async Task<FeatureFlagStatus> GetFeatureFlagAsync(ServiceEndpoint serviceEndpoint, string featureFlagName)
+        {
+            ArgUtil.NotNull(serviceEndpoint, nameof(serviceEndpoint));
+            ArgUtil.NotNull(featureFlagName, nameof(featureFlagName));
+            try
+            {
+                var accessTokenFF = serviceEndpoint?.Authorization?.Parameters["AccessToken"] ?? null;
+
+                var url = $"{serviceEndpoint.Url}_apis/FeatureFlags/{featureFlagName}";
+                httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {accessTokenFF}");
+                var response = await httpClient.GetAsync(url);
+                if (response.IsSuccessStatusCode)
+                {
+                    return await response.Content.ReadFromJsonAsync<FeatureFlagStatus>();
+                }
+                else
+                {
+                    throw new Exception("Failded to retrive FeatureFlag status");
+                }
+            } catch(Exception)
```

Comment:
Please don't catch(Exception).  If there is a specific exception that needs to be handled, please handle it narrowly and specifically

Reply:
@merlynomsft fixed, thank you